lib/matplotlib/dates.py

```diff
@@ -775,6 +775,10 @@ def format_ticks(self, values):
         for level in range(5, -1, -1):
             if len(np.unique(tickdate[:, level])) > 1:
                 break
+            elif level == 0:
```

Comment:
I think we should just re-introduce level 6, and have proper format for that level.

Reply:
@leolchat did you have any interest in moving this forward, or should I convert to an issue.  I still think this should get its own level, versus assuming level==0 is level 5.

Reply:
if we reach here level == 0 without calling `break`, it means that every level tested has one unique value, which is what happens when the only differences between ticks are sub-second:
```
[ 2020-06-06 18:08:30.4,  2020-06-06 18:08:30.5,  2020-06-06 18:08:30.6,  2020-06-06 18:08:30.7]
```
(That is what the test I added is testing)

Reply:
If it was not subsecond, then it would be the unit above years, which doesn't exist in datetime.